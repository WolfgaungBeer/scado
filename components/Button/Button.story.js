import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import Button from './';
import LabeledIcon from '../LabeledIcon';


storiesOf('Button', module)
    .add('default', () => (
        <Button onClick={action('clicked')}>Click me!</Button>
    ))
    .add('with props', () => (
        <Button
            type={text('type', 'flat')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            onClick={action('clicked')}
        >
            Click me!
        </Button>
    ))
    .add('with a custom component', () => (
        <Button
            type={text('type', 'flat')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            onClick={action('clicked')}
        >
            <LabeledIcon icon="dashboard" label="CLICK ME" />
        </Button>
    ));
