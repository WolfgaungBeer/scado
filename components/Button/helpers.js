export const getFontColor = () => (props) => {
    switch (props.type) {
    case 'flat':
        return props.color ? props.theme.color[props.color] : props.theme.color.black;
    case 'raised':
        return props.color ? props.theme.color.white : props.theme.color.black;
    default:
        return props.color ? props.theme.color.white : props.theme.color.black;
    }
};

export const getColor = () => (props) => {
    if (props.type) {
        return (props.type === 'flat') ? 'transparent' : props.theme.color[props.color];
    }
    return props.theme.color[props.color] || props.theme.color[props.lightGray];
};

export const getBorderStyle = () => (props) => {
    if (props.type) {
        return (props.type === 'flat') ? 'none' : 'solid';
    }
    return 'solid' || 'none';
};

export const getIconColor = (type, color) => {
    switch (type) {
    case 'flat':
        return color || 'black';
    case 'raised':
        return color ? 'white' : 'black';
    default:
        return color ? 'white' : 'black';
    }
};
