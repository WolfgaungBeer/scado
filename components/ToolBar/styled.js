import Flex from '../Flex';

export const StyledToolBar = Flex.extend`
    width: 100%;
    height: ${props => props.theme.scale.xxxl};
    background-color: ${props => props.theme.color.primary};
`;

export const ChildrenWrapper = Flex.extend`
    height: 100%;
`;
