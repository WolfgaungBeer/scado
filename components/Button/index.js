import React from 'react';
import { string, func, node } from 'prop-types';
import StyledButton from './StyledButton';
import { getIconColor } from './helpers';
import Icon from '../Icon';


const propTypes = {
    type: string,
    color: string,
    scale: string,
    icon: string,
    onClick: func,
    children: node
};

const defaultProps = {
    type: undefined,
    color: undefined,
    scale: undefined,
    icon: undefined,
    onClick: undefined,
    children: undefined
};

const Button = ({ type, color, scale, icon, onClick, children }) => {

    const buttonType = type || 'default';
    const iconColor = getIconColor(type, color);
    const iconOnly = (icon && !children);

    return (
        <StyledButton type={buttonType} color={color} scale={scale} iconOnly={iconOnly} onClick={onClick}>
            <div>
                {icon && <Icon icon={icon} color={iconColor} scale={scale} />}
                {children}
            </div>
        </StyledButton>
    );
};

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
