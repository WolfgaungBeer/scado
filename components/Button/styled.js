import styled from 'styled-components';
import { getBackgroundColor, getBorder, getShadow, getFontColor } from './helpers';

export const StyledButton = styled.button`
    width: 100%;
    cursor: pointer;
    background-color: ${getBackgroundColor()};
    ${getBorder()};
    ${getShadow()};
    ${getFontColor()}
    font-size: ${props => props.theme.scale[props.scale]};
    padding: 0.5rem 1rem;
    &:focus {
        outline: 0;
    }
`;
