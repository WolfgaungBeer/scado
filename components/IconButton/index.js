import React from 'react';
import { string, bool, func } from 'prop-types';
import { StyledButton } from './styled';
import Icon from '../Icon';

const propTypes = {
    icon: string.isRequired,
    color: string,
    scale: string,
    loading: bool,
    onClick: func.isRequired,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    loading: undefined,
};

const IconButton = ({ icon, color, scale, loading, onClick }) => (
    <StyledButton onClick={onClick}>
        <Icon icon={icon} color={color} scale={scale} loading={loading} />
    </StyledButton>
);

IconButton.propTypes = propTypes;
IconButton.defaultProps = defaultProps;

export default IconButton;
