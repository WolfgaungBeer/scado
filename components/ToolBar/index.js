import React from 'react';
import { node } from 'prop-types';
import { StyledToolBar, ChildrenWrapper } from './styled';


const propTypes = {
    children: node,
    actionButtons: node,
};

const defaultProps = {
    children: undefined,
    actionButtons: undefined,
};

const ToolBar = ({ children, actionButtons }) => (
    <StyledToolBar alignItems="center" justifyContent="space-between">
        <ChildrenWrapper alignItems="center">{children}</ChildrenWrapper>
        {actionButtons && <ChildrenWrapper alignItems="center">{actionButtons}</ChildrenWrapper>}
    </StyledToolBar>
);

ToolBar.propTypes = propTypes;
ToolBar.defaultProps = defaultProps;

export default ToolBar;
