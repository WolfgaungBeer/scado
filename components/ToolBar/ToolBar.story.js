import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import ToolBar from './';
import IconButton from '../IconButton';
import Text from '../Text';

const ActionButtons = () => (
    <div>
        <IconButton icon="search" color="white" scale="l" onClick={action('action button 1 clicked')} />
        <IconButton icon="settings" color="white" scale="l" onClick={action('action button 2 clicked')} />
        <IconButton icon="keyboard_arrow_down" color="white" scale="l" onClick={action('action button 3 clicked')} />
    </div>
);

storiesOf('ToolBar', module)
    .add('default', () => (
        <ToolBar actionButtons={<ActionButtons />}>
            <IconButton icon="dashboard" color="white" scale="xl" onClick={action('menu clicked')} />
            <Text.Span color="white" scale="l" onClick={action('text clicked')}>Massiv App</Text.Span>
        </ToolBar>
    ));
