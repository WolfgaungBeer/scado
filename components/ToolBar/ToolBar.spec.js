import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import ToolBar from './';
import IconButton from '../IconButton';
import Button from '../Button';

const click = () => {};

const ActionButtons = () => (
    <div>
        <IconButton icon="search" color="white" scale="l" onClick={click} />
        <IconButton icon="settings" color="white" scale="l" onClick={click} />
        <IconButton icon="keyboard_arrow_down" color="white" scale="l" onClick={click} />
    </div>
);

it('renders the ToolBar Component correctly', () => {
    const tree = renderer.create(
        <Wrapper>
            <ToolBar actionButtons={<ActionButtons />}>
                <IconButton icon="dashboard" color="white" scale="xl" onClick={click} />
                <Button buttonType="flat" color="white" scale="l" onClick={click}>Massiv App</Button>
            </ToolBar>
        </Wrapper>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
