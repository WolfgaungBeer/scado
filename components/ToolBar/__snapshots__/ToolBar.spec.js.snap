// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ToolBar Component correctly 1`] = `
<div
  className="sc-bwzfXH iVCEtc"
>
  <div
    className="sc-htpNat btXhPq"
  >
    <button
      className="sc-bxivhb cjQIgV"
      onClick={[Function]}
    >
      <i
        className="material-icons  sc-ifAKCX kgnKXz"
        color="white"
        scale="xl"
      >
        dashboard
      </i>
    </button>
    <button
      className="sc-EHOje ftkoff"
      color="white"
      onClick={[Function]}
      scale="l"
      type="flat"
    >
      Massiv App
    </button>
  </div>
  <div
    className="sc-htpNat btXhPq"
  >
    <div>
      <button
        className="sc-bxivhb cjQIgV"
        onClick={[Function]}
      >
        <i
          className="material-icons  sc-ifAKCX cpnXXE"
          color="white"
          scale="l"
        >
          search
        </i>
      </button>
      <button
        className="sc-bxivhb cjQIgV"
        onClick={[Function]}
      >
        <i
          className="material-icons  sc-ifAKCX cpnXXE"
          color="white"
          scale="l"
        >
          settings
        </i>
      </button>
      <button
        className="sc-bxivhb cjQIgV"
        onClick={[Function]}
      >
        <i
          className="material-icons  sc-ifAKCX cpnXXE"
          color="white"
          scale="l"
        >
          keyboard_arrow_down
        </i>
      </button>
    </div>
  </div>
</div>
`;
